pkg/commands/volume.go

```diff
@@ -48,15 +48,15 @@ func (v *VolumeCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile.
 	for _, volume := range resolvedVolumes {
 		var x struct{}
 		existingVolumes[volume] = x
-		err := util.AddVolumePathToWhitelist(volume)
+		err := util.AddPathToVolumeWhitelist(volume)
 		if err != nil {
 			return err
 		}
 
 		// Only create and snapshot the dir if it didn't exist already
 		if _, err := os.Stat(volume); os.IsNotExist(err) {
```

Comment:
[Severity-high]: By setting `v.snapshotFiles` to a single-element slice on every volume addition, we might be discarding previously added volumes. Is this the intended behavior? It seems like appending to the existing `v.snapshotFiles` might be more appropriate.
```suggestion
v.snapshotFiles = append(v.snapshotFiles, volume)
```